### ISSUE TITLE
Can't import recording appender from typescript

### ISSUE THREAD

**Issue by daniel-beard** [log4js-node/log4js-node#967]
Doesn't look like the types for the recording appender are defined maybe?

```typescript
import * as recording from "log4js/lib/appenders/recording";
```

Outputs:
```log
TSError: TypeScript diagnostics (customize using `[jest-config].globals.ts-jest.diagnostics` option):
src/utils/logging.ts:2:27 - error TS7016: Could not find a declaration file for module 'log4js/lib/appenders/recording'. '/project/node_modules/log4js/lib/appenders/recording.js' implicitly has an 'any' type.
  Try `npm install @types/log4js` if it exists or add a new declaration (.d.ts) file containing `declare module 'log4js/lib/appenders/recording';`

2 import { recording } from 'log4js/lib/appenders/recording';
                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Is there a way for me to import this appender in typescript?

**Comment by nomiddlename** [comment]
It's defined in `types/log4js.d.ts` but that's from the perspective of using `log4js.configure` with a recording appender in the config rather than importing it directly in a test. I don't use typescript, so I rely on users to help me with the types - so if you feel like making a PR with the changes you need that would be awesome.

**Comment by daniel-beard** [comment]
I have not been able to figure out how to do this yet in the source project, however, so far have been able to use this in my project under `typings/recording.d.ts`:

```typescript
declare module 'log4js/lib/appenders/recording' {
  export function replay(): any[];
  export function reset(): void;
}
```

Which isn't great, but works. Will keep investigating, but am pretty new to TS.

**Comment by polo-language** [comment]
TypeScript bindings to work with the recording are available in PR #1103. Let me know if it looks good to you.
